### DISCUSSION TITLE
How does AIR call a webcam?

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#2723] [Q&A]
I know that Camera can call the USB camera of this machine, but now there is a requirement to call the webcam through an IP. Are there any relevant libraries and methods that can be called?
For example, some webcam brands may prompt for an RTSP address, and I don't know what method to use to call it.
rtsp://admin:hik12345@192.168.8.125:554/h264/ch33/main/av_stream
Thank you!

**Comment by ajwfrost** [comment]
Hi

AIR accesses cameras via the standard drivers available on an operating system, so for an IP-based camera this could depend.. if there are any drivers that are able to make the camera appear in normal apps then it may also work with AIR (so e.g. if you open up Zoom or a similar tool, do you get the option to select the camera there?)

We'd done some work with an IP-based camera a few years ago which used RTSP, and yes for this the camera just provided an RTSP server which we had to connect to and stream the video down and then push into AIR using the `NetStream.appendBytes()` function. But that involved developing a basic RTSP client which we did in C++ and then compiled that to SWF using CrossBridge..

Our intention for the new air.media package is that we can have flexibility here for RTSP based sources and other such ingest, with the video stream then being able to pass in to the AIR decoder and display. And one of the key parts is to ensure that third parties are able to create elements here as ANEs or similar, so that someone else could write a handler for RTSP streams and it wouldn't be gated by us!

thanks
